operators/pkg/forge/virtualmachines.go

```diff
@@ -51,13 +51,16 @@ func VirtualMachineSpec(instance *clv1alpha2.Instance, environment *clv1alpha2.E
 
 // VirtualMachineInstanceSpec forges the specification of a Kubevirt VirtualMachineInstance
 // object representing the definition of the VMI corresponding to a non-persistent CrownLabs Environment.
-func VirtualMachineInstanceSpec(instance *clv1alpha2.Instance, environment *clv1alpha2.Environment) virtv1.VirtualMachineInstanceSpec {
+func VirtualMachineInstanceSpec(instance *clv1alpha2.Instance, environment *clv1alpha2.Environment, includeCloudInit bool) virtv1.VirtualMachineInstanceSpec {
+	volumes := []virtv1.Volume{VolumeRootDisk(instance, environment)}
+	// Attach cloudinit volume on non-restricted environments
+	if includeCloudInit {
+		volumes = append(volumes, VolumeCloudInit(NamespacedName(instance).Name))
+	}
+
```

Comment:
Would personally maintain the previous structure, moving this to an appropriate function `Volumes` (to simplify testing)